### PR TITLE
DAOS-623 vos: Fix a memory leak

### PR DESCRIPTION
**Pull request by jolivier23**
Fixes a memory leak in vos IO test
Adds a few instances of DAOS_ON_VALGRIND to speed up valgrind runs
Fix a suppression for gc when run standalone

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Review by NiuYawei — APPROVED** [review]

File: src/vos/tests/vts_io.c [hunk 1]
```diff
@@ -911,6 +911,7 @@ io_obj_cache_test(void **state)
 	rc = vos_pool_destroy(po_name, pool_uuid);
 	assert_int_equal(rc, 0);
 	vos_obj_cache_destroy(occ);
+	free(po_name);
```
Comment (ashleypittman) [inline src/vos/tests/vts_io.c:914]: Is this the leak that the suppression is remove for?

**Review by ashleypittman — COMMENTED** [review]
There is a race conditional and occasional failure in the unit test/memcheck tests, I would prefer to delay changes until that can be identified/resolved to avoid making that issue more difficult.

**Review by Nasf-Fan — APPROVED** [review]

File: src/vos/tests/vts_io.c (hunk 1 above)
Comment (jolivier23) [reply src/vos/tests/vts_io.c:914]: Yes

**Comment by jolivier23** [comment]
> There is a race conditional and occasional failure in the unit test/memcheck tests, I would prefer to delay changes until that can be identified/resolved to avoid making that issue more difficult.

We've identified the issue.  The script that invokes run_test.sh moves the files after invoking it but also uses set -e.  Marcela is going to upload a patch to produce the files in the right place to start with which should fix that issue.

This PR doesn't really change or fix anything that would normally fail.  It fixes suppressions for running individual tests locally (e.g. vos_tests -i rather than -A)

**Comment by ashleypittman** [comment]
> We've identified the issue. The script that invokes run_test.sh moves the files after invoking it but also uses set -e. Marcela is going to upload a patch to produce the files in the right place to start with which should fix that issue.

This seems to be a recurring failure mode for us for some reason.  We should always be creating the files in place unless there's a good reason not to.

> This PR doesn't really change or fix anything that would normally fail. It fixes suppressions for running individual tests locally (e.g. vos_tests -i rather than -A)

I'll approve this then.

**Review by ashleypittman — APPROVED** [review]

**Pull request merged**